spec/numberformat.html

```diff
@@ -23,26 +30,43 @@ <h1>SetNumberFormatDigitOptions ( _intlObj_, _options_, _mnfdDefault_, _mxfdDefa
         1. Let _mxsd_ be ? Get(_options_, *"maximumSignificantDigits"*).
         1. Set _intlObj_.[[MinimumIntegerDigits]] to _mnid_.
         1. If _mnsd_ is not *undefined* or _mxsd_ is not *undefined*, then
-          1. Set _intlObj_.[[RoundingType]] to ~significantDigits~.
-          1. Let _mnsd_ be ? DefaultNumberOption(_mnsd_, 1, 21, 1).
-          1. Let _mxsd_ be ? DefaultNumberOption(_mxsd_, _mnsd_, 21, 21).
+          1. Let _hasSd_ be *true*.
+        1. Else,
+          1. Let _hasSd_ be *false*.
+        1. If _mnfd_ is not *undefined* or _mxfd_ is not *undefined*, then
+          1. Let _hasFd_ be *true*.
+        1. Else,
+          1. Let _hasFd_ be *false*.
+        1. Let _needSd_ be _hasSd_.
+        1. If _hasSd_ is *false* and (_hasFd_ is *true* or _notation_ is not *"compact"*), then
+          1. Let _needFd_ be *true*.
+        1. Else,
+          1. Let _needFd_ be *false*.
```

Comment:
I don't think we express things with parentheses like this in spec text.
```suggestion
        1. If _hasSd_ is *true*, or _hasFd_ is *false* and _notation_ is *"compact"*, then
          1. Let _needFd_ be *false*.
        1. Else,
          1. Let _needFd_ be *true*.
```